### PR TITLE
COMMERCE-2316

### PR DESCRIPTION
**Pull request by alessiorendina**

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test
tests should be fixed in upstream now

**Review by ethanbustad — APPROVED** [review]

File: commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java [hunk 1]
```diff
@@ -418,7 +418,34 @@ public CommerceOrder checkoutCommerceOrder(
...
+
+					@Override
+					public Void call() throws Exception {
+						Message message = new Message();
+
+						message.put(
+							"cpInstanceId",
+							commerceOrderItem.getCPInstanceId());
+
+						MessageBusUtil.sendMessage(
```
Comment (marco-leo) [inline commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java]: @Commando18 This is a check that make sense to put in the consume quantity method of the inventory engine, and I would like to get an integration test with this fix that cover it

**Review by marco-leo — CHANGES_REQUESTED** [review]

**Comment by marco-leo** [comment]
ci:test

**Comment by alessiorendina** [comment]
ci:test

File: commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java [hunk 2]
```diff
@@ -1832,6 +1838,61 @@ protected void validatePurchaseOrderNumber(String purchaseOrderNumber)
...
+			commerceOrder.getCommerceOrderItems();
+
+		for (CommerceOrderItem commerceOrderItem : commerceOrderItems) {
+			Map<String, String> context = new HashMap<>();
+
+			context.put(
+				"OrderId ",
+				String.valueOf(commerceOrderItem.getCommerceOrderId()));
+			context.put(
+				"OrderItemId ",
```
Comment (ethanbustad) [inline commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java:1854]: should these keys have a trailing space in them?

Comment (ethanbustad) [reply commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java:1854]: Oh... looks like they had a trailing space before? Weird, but alright.

**Review by ethanbustad — APPROVED** [review]

**Pull request merged**